mobile/android/Interface.cpp

```diff
@@ -242,16 +242,24 @@ JNIEXPORT jlong JNICALL Java_org_iota_mobile_Interface_bundle_1miner_1mine(
   byte_t const* bundleNormalizedMax = (byte_t*)env->GetByteArrayElements(jbundleNormalizedMax, 0);
   trit_t* essence = (trit_t*)env->GetByteArrayElements(jessence, 0);
 
-  ret = bundle_miner_mine(bundleNormalizedMax, jsecurity, essence, jessenceLength, jcount, jnprocs, jminingThreshold,
-                          jfullySecure, &index);
+  bundle_miner_ctx_t* ctxs = NULL;
+  size_t num_ctxs = 0;
+  bool found_optimal_index = false;
+
+  bundle_miner_allocate_ctxs(jnprocs, &ctxs, &num_ctxs);
+
+  ret = bundle_miner_mine(bundleNormalizedMax, jsecurity, essence, jessenceLength, jcount, jminingThreshold,
+                          jfullySecure, &index, ctxs, num_ctxs, &found_optimal_index);
 
   env->ReleaseByteArrayElements(jbundleNormalizedMax, (jbyte*)bundleNormalizedMax, 0);
   env->ReleaseByteArrayElements(jessence, (jbyte*)essence, 0);
 
   if (ret != RC_OK) {
+    bundle_miner_deallocate_ctxs(&ctxs);
```

Comment:
You deallocate in both case so you can do it before the if

Reply:
Whoops, fixed in 0325f84